bin/omarchy-update-git

```diff
@@ -1,5 +1,15 @@
 #!/bin/bash
 
+trap 'echo -e "\n\e[33mUpdate interrupted.\e[0m"; exit 130' INT TERM HUP
+
 echo -e "\e[32mUpdate Omarchy\e[0m"
 git -C $OMARCHY_PATH pull --autostash
-git -C $OMARCHY_PATH diff --check || git -C $OMARCHY_PATH reset --merge
+
+if ! git -C $OMARCHY_PATH diff --check; then
+  if gum confirm "Omarchy update requires removing your local changes. Continue?"; then
+    git -C $OMARCHY_PATH reset --merge
```

Comment:
As mentioned in https://github.com/basecamp/omarchy/pull/1312#issuecomment-3240388957, if the user had conflicts, those changes will still be in the stash - so "removing your local changes" sounds scarier than it is.

Anyway, if you want to introduce a `gum confirm`, consider instead doing so before we even execute the `git pull --autostash`, e.g. if the working directory is dirty, say something like "Your omarchy directory is dirty; we will autostash these changes before pulling. Proceed with the update?" (If they proceed, but then decide to Ctrl+C, it's probably fine to let them deal with manually restoring from the stash - not sure the `trap` is needed.)